src/main/java/games/strategy/triplea/delegate/AbstractPlaceDelegate.java

```diff
@@ -509,12 +509,6 @@ protected String moveAirOntoNewCarriers(final Territory at, final Territory prod
       return null;
     }
     final List<Unit> fighters = producer.getUnits().getMatches(ownedFighters);
-    while (fighters.size() > 0 && AirMovementValidator.carrierCost(fighters) > capacity) {
-      fighters.remove(0);
-    }
-    if (fighters.size() == 0) {
-      return null;
-    }
```

Comment:
This is removed so all available fighters are passed to the selection method instead of filtering it randomly to a max of carrier cost.